client/state/sites/selectors.js

```diff
@@ -123,3 +123,46 @@ export function getSiteByUrl( state, url ) {
 
 	return site;
 }
+
+/**
+ * Returns a site's plan object by site ID.
+ *
+ * The difference between this selector and sites/plans/getPlansBySite is that the latter selectors works
+ * with the /sites/$site/plans endpoint while the former selectors works with /sites/$site endpoint.
+ * Query these endpoints to see if you need the first or the second one.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @return {?Object}        Site's plan object
+ */
+export function getSitePlan( state, siteId ) {
```

Comment:
I think it's useful to have these specialized selectors, as I'm not forced to remember what is available, or what the data structure looks like, or be forced to update my call should the data structure change. (Esp if we have global selectors)

Internally we can still have a method like `getSiteProperty( state, siteId, property )` where the specialized selector might look like:

```javascript
getSitePlan( state, sideId ) {
    return getSiteProperty( state, siteId, 'plan' );
}
```